Packs/Palo_Alto_Networks_WildFire/Integrations/Palo_Alto_Networks_WildFire_v2/Palo_Alto_Networks_WildFire_v2.py

```diff
@@ -700,163 +829,147 @@ def create_file_report(file_hash: str, reports, file_info, format_: str = 'xml',
+        return command_results, entry_context['Status']
+
+
+def wildfire_get_report_command(args):
+    """
+    Args:
+        args: the command arguments arguments from demisto.args(), including url or file hash (sha256 or md5) to query on
+
+    Returns:
+        A single or list of CommandResults, and the status of the reports of the url or file of interest.
+        Note that the status is only used for the polling sequence, where the command will always receive a single
+        file or url. Hence, when running this command via the polling sequence, the CommandResults list will contain a
+        single item, and the status will represent that result's status.
+
+    """
+    command_results_list = []
+    urls = argToList(args.get('url', ''))
+    if 'sha256' in args:
+        sha256 = args.get('sha256')
+    elif 'hash' in args:
+        sha256 = args.get('hash')
     else:
         sha256 = None
-    md5 = demisto.args().get('md5', None)
-    if urls:
-        inputs = urls
-        url_report = True
-    else:
-        inputs = hash_args_handler(sha256, md5)
-        url_report = False
+    md5 = args.get('md5')
+    inputs = urls if urls else hash_args_handler(sha256, md5)
 
-    verbose = demisto.args().get('verbose', 'false').lower() == 'true'
-    format_ = demisto.args().get('format', 'xml')
     for element in inputs:
-        if url_report:
-            url, report = wildfire_get_url_report(element)
-            if url is not None:
-                headers = ['sha256', 'type', 'verdict', 'iocs']
-                human_readable = tableToMarkdown(f'Wildfire URL report for {url}', t=report, headers=headers,
-                                                 removeNull=True)
-                entry_context = {"WildFire.Report(val.URL == obj.URL)": report}
-                return_outputs(human_readable, entry_context, report)
-        else:
-            ioc, report, file_info = wildfire_get_file_report(element)
-            if ioc is not None:
-                create_file_report(ioc, report, file_info, format_, verbose)
+        command_results, status = wildfire_get_url_report(element) if urls else wildfire_get_file_report(element, args)
+        command_results_list.append(command_results)
+
+    return command_results_list, status
 
 
-def wildfire_file_command():
-    inputs = file_args_handler(demisto.args().get('file'), demisto.args().get('md5'), demisto.args().get('sha256'))
+def wildfire_file_command(args):
+    inputs = file_args_handler(args.get('file'), args.get('md5'), args.get('sha256'))
+    command_results_list = []
     for element in inputs:
         if sha1Regex.match(element):
             demisto.results({
                 'Type': 11,
                 'Contents': 'WildFire file hash reputation supports only MD5, SHA256 hashes',
                 'ContentsFormat': formats['text']
             })
+            return
```

Comment:
why do we stop the iteration and not continue to the next elements?

Reply:
shouldn't be there, removed